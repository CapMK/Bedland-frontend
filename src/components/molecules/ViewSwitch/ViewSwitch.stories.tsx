import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import ViewSwitch from './ViewSwitch';

const meta = {
  title: 'Molecules/ViewSwitch',
  component: ViewSwitch,
  tags: ['autodocs'],

  parameters: {
    backgrounds: { default: 'dark' },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof ViewSwitch>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Main: Story = {};
