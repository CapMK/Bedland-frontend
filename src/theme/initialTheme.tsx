import { Theme } from './theme.types';

const initialTheme: Theme = {
  palette: {
    sidebarTabsPressed: '#919AFF',
    blueDark: '#4553DD',
    blueLight: '#6573FF',
    orangeLight: '#FF7C33',
    orangeDark: '#F56513',
    light: '#FCFEFF',
    greyLight: '#F5F6F8',
    greyDark: '#767676',
    greenLight: '#83E57A',
    greenDark: '#30BA57',
    redLight: '#FF4438',
    sidebarTabs: '#CACACA',
    redDark: '#CF0000',
    sidebar: '#0A1122',
    footer: '#3F4476',
    white: '#fff',
    inputGrey: '#4B4B4B',
    widgetsShadows: '#D9D9D9',
    text: '#1B1A1A',
    yellow: '#FFD541',
  },
  typography: {
    header1: {
      fontFamily: 'Poppins, sans-serif',
      fontWeight: 600,
      fontSize: '2rem',
      letterSpacing: '2px',
      lineHeight: 1.2,
    },
    header2: {
      fontFamily: 'Poppins, sans-serif',
      fontWeight: 700,
      fontSize: '1.5rem',
      lineHeight: 1.4,
    },
    subtitles: {
      fontFamily: 'Poppins, sans-serif',
      fontWeight: 500,
      fontSize: '1.25rem',
      lineHeight: 1.4,
      letterSpacing: '1px',
    },
    header3: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 700,
      fontSize: '1.25rem',
      lineHeight: 1.4,
      letterSpacing: '1px',
    },
    header4: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontSize: '1rem',
      fontWeight: 600,
      lineHeight: 1.4,
      letterSpacing: '1px',
    },
    header5: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 600,
      fontSize: '0.875rem',
      lineHeight: 1.4,
      letterSpacing: '0',
    },
    subHeader: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 400,
      fontSize: '0.75rem',
      lineHeight: 1.4,
      letterSpacing: '0',
    },
    paragraph: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: 1.6,
      letterSpacing: '0',
    },
    dataInputsAndTooltips: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 400,
      fontSize: '0.875rem',
      lineHeight: 1.4,
      letterSpacing: '1px',
    },
    unerlined: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 400,
      fontSize: '0.875rem',
      lineHeight: 1.6,
      textDecoration: 'underline',
    },
    captions: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 600,
      fontSize: '0.75rem',
      lineHeight: 1.6,
      letterSpacing: '1px',
    },
  },
};

export default initialTheme;
