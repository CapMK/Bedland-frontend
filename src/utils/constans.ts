export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
