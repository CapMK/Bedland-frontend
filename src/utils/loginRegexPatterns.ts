import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
