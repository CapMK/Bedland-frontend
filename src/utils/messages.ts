const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
