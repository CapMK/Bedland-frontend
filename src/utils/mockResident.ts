const mockResidentList = [
  {
    id: 1,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'iSutherland',
    password: 'IsABEl',
    name: 'Isabel',
    lastName: 'Sutherland',
    email: 'isabel.sutherland@email.com',
    phoneNumber: '123456789',
    flatId: 1,
    owner: true,
  },
  {
    id: 2,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'clWeiss',
    password: 'ClWeiss',
    name: 'Claude',
    lastName: 'Weiss',
    email: 'Claude.Weiss@email.com',
    phoneNumber: '123456789',
    flatId: 2,
    owner: false,
  },
];

export default mockResidentList;
