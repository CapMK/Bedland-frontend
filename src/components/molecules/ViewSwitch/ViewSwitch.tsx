import Icon from 'components/atoms/Icon/Icon';
import { useState } from 'react';
import * as Styled from './ViewSwitch.styled';

const ViewSwitch = () => {
  const [checked, setChecked] = useState<boolean>(false);
  return (
    <Styled.Switcher data-testid='view-switch'>
      <Styled.CustomInput type='checkbox' onClick={() => setChecked(!checked)} />
      <Styled.Slider checked={checked} data-testid='view-switch-slider'>
        <Icon name='listbox' size={16} color={checked ? 'greyDark' : 'blueDark'} isActive={false} />
        <Icon
          name='buildingPlan'
          size={16}
          color={checked ? 'blueDark' : 'greyDark'}
          isActive={false}
        />
      </Styled.Slider>
    </Styled.Switcher>
  );
};

export default ViewSwitch;
