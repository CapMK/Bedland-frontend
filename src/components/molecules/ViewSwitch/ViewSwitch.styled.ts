import styled from 'styled-components';
import initialTheme from 'theme/initialTheme';

export const Switcher = styled.label`
  display: flex;
  height: 2.5rem;
  position: relative;
  width: 5rem;
`;

export const Slider = styled.span<{ checked: boolean }>`
  -webkit-transition: 0.4s;
  align-items: center;
  background-color: ${initialTheme.palette.sidebarTabs};
  border-radius: 8px;
  bottom: 0;
  cursor: pointer;
  display: flex;
  gap: 1.5rem;
  justify-content: center;
  position: absolute;
  top: 0;
  transition: 0.4s;
  width: 100%;
  z-index: 2;
  &:before {
    -webkit-transition: 0.4s;
    background-color: ${initialTheme.palette.light};
    border-radius: 8px;
    bottom: 0;
    content: '';
    height: 2.5rem;
    left: 0;
    position: absolute;
    transform: ${({ checked }) => (checked ? 'translateX(40px)' : 'translateX(0px)')};
    transition: 0.4s;
    width: 2.5rem;
    z-index: -1;
  }
`;

export const CustomInput = styled.input`
  height: 0;
  opacity: 0;
  width: 0;
`;
