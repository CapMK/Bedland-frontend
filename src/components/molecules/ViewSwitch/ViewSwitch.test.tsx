import { renderWithProviders } from 'utils/test.utils';
import { fireEvent, screen, cleanup } from '@testing-library/react';
import initialTheme from 'theme/initialTheme';
import ViewSwitch from './ViewSwitch';

describe('ViewSwitch', () => {
  afterEach(cleanup);
  const page = () => renderWithProviders(<ViewSwitch />, {});
  it('renders correctly', () => {
    expect(page()).toMatchSnapshot();
  });

  it('View Switch correctly changes after clicking', async () => {
    page();

    const viewSwitch = screen.getByTestId('view-switch');
    const viewSwitchSlider = screen.getByTestId('view-switch-slider');
    const listboxIcon = screen.getByTestId('listbox-icon').querySelector('path');
    const buildingPlanIcon = screen.getByTestId('buildingPlan-icon').querySelector('path');

    expect(listboxIcon?.getAttribute('stroke')).toBe(initialTheme.palette.blueDark);
    expect(buildingPlanIcon?.getAttribute('stroke')).toBe(initialTheme.palette.greyDark);
    expect(viewSwitchSlider).toHaveStyleRule('transform', 'translateX(0px)', {
      modifier: ':before',
    });

    await fireEvent.click(viewSwitch);
    expect(listboxIcon?.getAttribute('stroke')).toBe(initialTheme.palette.greyDark);
    expect(buildingPlanIcon?.getAttribute('stroke')).toBe(initialTheme.palette.blueDark);
    expect(viewSwitchSlider).toHaveStyleRule('transform', 'translateX(40px)', {
      modifier: ':before',
    });

    await fireEvent.click(viewSwitch);
    expect(listboxIcon?.getAttribute('stroke')).toBe(initialTheme.palette.blueDark);
    expect(buildingPlanIcon?.getAttribute('stroke')).toBe(initialTheme.palette.greyDark);
    expect(viewSwitchSlider).toHaveStyleRule('transform', 'translateX(0px)', {
      modifier: ':before',
    });
  });
});
