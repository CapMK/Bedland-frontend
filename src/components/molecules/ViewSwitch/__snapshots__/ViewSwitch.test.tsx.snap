// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ViewSwitch > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.5rem;
  position: relative;
  width: 5rem;
}

.c2 {
  -webkit-transition: 0.4s;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CACACA;
  border-radius: 8px;
  bottom: 0;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  width: 100%;
  z-index: 2;
}

.c2:before {
  -webkit-transition: 0.4s;
  background-color: #FCFEFF;
  border-radius: 8px;
  bottom: 0;
  content: '';
  height: 2.5rem;
  left: 0;
  position: absolute;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  -webkit-transition: 0.4s;
  transition: 0.4s;
  width: 2.5rem;
  z-index: -1;
}

.c1 {
  height: 0;
  opacity: 0;
  width: 0;
}

<body>
    <div>
      <label
        class="c0"
        data-testid="view-switch"
      >
        <input
          class="c1"
          type="checkbox"
        />
        <span
          class="c2"
          data-testid="view-switch-slider"
        >
          <svg
            data-testid="listbox-icon"
            fill="none"
            height="16"
            viewBox=" 0 0 18 18"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              id="mask-listbox"
            />
            <path
              d="M8 5H13M13 9H8M13 13H8M5 5H5.001M5 9H5.001M5 13H5.001M3 17H15C15.5304 17 16.0391 16.7893 16.4142 16.4142C16.7893 16.0391 17 15.5304 17 15V3C17 2.46957 16.7893 1.96086 16.4142 1.58579C16.0391 1.21071 15.5304 1 15 1H3C2.46957 1 1.96086 1.21071 1.58579 1.58579C1.21071 1.96086 1 2.46957 1 3V15C1 15.5304 1.21071 16.0391 1.58579 16.4142C1.96086 16.7893 2.46957 17 3 17Z"
              stroke="#4553DD"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
          <svg
            data-testid="buildingPlan-icon"
            fill="none"
            height="16"
            viewBox=" 0 0 18 18"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              id="mask-buildingPlan"
            />
            <path
              d="M17 16V7.5C17 7.34475 16.9639 7.19164 16.8944 7.05279C16.825 6.91393 16.7242 6.79315 16.6 6.7L9.6 1.45C9.4269 1.32018 9.21637 1.25 9 1.25C8.78363 1.25 8.5731 1.32018 8.4 1.45L1.4 6.7C1.2758 6.79315 1.175 6.91393 1.10557 7.05279C1.03614 7.19164 1 7.34475 1 7.5V16C1 16.2652 1.10536 16.5196 1.29289 16.7071C1.48043 16.8946 1.73478 17 2 17H16C16.2652 17 16.5196 16.8946 16.7071 16.7071C16.8946 16.5196 17 16.2652 17 16Z"
              stroke="#767676"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.67"
            />
          </svg>
        </span>
      </label>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.5rem;
  position: relative;
  width: 5rem;
}

.c2 {
  -webkit-transition: 0.4s;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CACACA;
  border-radius: 8px;
  bottom: 0;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  width: 100%;
  z-index: 2;
}

.c2:before {
  -webkit-transition: 0.4s;
  background-color: #FCFEFF;
  border-radius: 8px;
  bottom: 0;
  content: '';
  height: 2.5rem;
  left: 0;
  position: absolute;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  -webkit-transition: 0.4s;
  transition: 0.4s;
  width: 2.5rem;
  z-index: -1;
}

.c1 {
  height: 0;
  opacity: 0;
  width: 0;
}

<div>
    <label
      class="c0"
      data-testid="view-switch"
    >
      <input
        class="c1"
        type="checkbox"
      />
      <span
        class="c2"
        data-testid="view-switch-slider"
      >
        <svg
          data-testid="listbox-icon"
          fill="none"
          height="16"
          viewBox=" 0 0 18 18"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="mask-listbox"
          />
          <path
            d="M8 5H13M13 9H8M13 13H8M5 5H5.001M5 9H5.001M5 13H5.001M3 17H15C15.5304 17 16.0391 16.7893 16.4142 16.4142C16.7893 16.0391 17 15.5304 17 15V3C17 2.46957 16.7893 1.96086 16.4142 1.58579C16.0391 1.21071 15.5304 1 15 1H3C2.46957 1 1.96086 1.21071 1.58579 1.58579C1.21071 1.96086 1 2.46957 1 3V15C1 15.5304 1.21071 16.0391 1.58579 16.4142C1.96086 16.7893 2.46957 17 3 17Z"
            stroke="#4553DD"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
        <svg
          data-testid="buildingPlan-icon"
          fill="none"
          height="16"
          viewBox=" 0 0 18 18"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="mask-buildingPlan"
          />
          <path
            d="M17 16V7.5C17 7.34475 16.9639 7.19164 16.8944 7.05279C16.825 6.91393 16.7242 6.79315 16.6 6.7L9.6 1.45C9.4269 1.32018 9.21637 1.25 9 1.25C8.78363 1.25 8.5731 1.32018 8.4 1.45L1.4 6.7C1.2758 6.79315 1.175 6.91393 1.10557 7.05279C1.03614 7.19164 1 7.34475 1 7.5V16C1 16.2652 1.10536 16.5196 1.29289 16.7071C1.48043 16.8946 1.73478 17 2 17H16C16.2652 17 16.5196 16.8946 16.7071 16.7071C16.8946 16.5196 17 16.2652 17 16Z"
            stroke="#767676"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.67"
          />
        </svg>
      </span>
    </label>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
